protocol/contracts/templegold/StakedTempleVoteToken.sol

```diff
@@ -0,0 +1,204 @@
+    
+    constructor(
+        address _initialRescuer,
+        address _initialExecutor,
+        address _staking,
+        string memory _name,
+        string memory _symbol
+    ) ERC20(_name, _symbol) TempleElevatedAccess(_initialRescuer, _initialExecutor) {
+        staking = _staking;
+    }
+
+    /**  
+     * @notice Set staking contract
+     * @param _staking Staking contract
+     */
+    function setStaking(address _staking) external override onlyElevatedAccess {
+        if (_staking == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        staking = _staking;
+        emit StakingSet(_staking);
+    }
+
+    /**  
+     * @notice Set authorized contract
+     * @dev Aucthorized contract is allowed to transfer/burn vote token
+     * @param _authority Contract
+     * @param _authorized Bool if authorized
+     */
+    function setAuthorized(address _authority, bool _authorized) external override onlyElevatedAccess {
+        if (_authority == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        authorized[_authority] = _authorized;
+        emit AuthoritySet(_authority, _authorized);
+    }
+
+    /**  
+     * @notice Mint vote token
+     * @dev Only authorized can mint
+     * @param to Recipient
+     * @param amount Amount to mint
+     */
+    function mint(address to, uint256 amount) external override onlyAuthorized whenNotPaused {
+        _mint(to, amount);
+    }
+
+    /// @notice override to stop holders burning vote tokens themselves
+    function burn(uint256 /*amount*/) public virtual override(IStakedTempleVoteToken, ERC20Burnable) {
+        /// @dev not implemented
+        revert NotImplemented();
+    }
+
+    /**
+     * @dev Destroys a `value` amount of tokens from `account`, deducting from
+     * the caller's allowance.
+     *
+     * See {ERC20-_burn} and {ERC20-allowance}.
+     *
+     * Requirements:
+     *
+     * - the caller must have allowance for ``accounts``'s tokens of at least
+     * `value`.
+     */
+    function burnFrom(address account, uint256 value) public virtual override(IStakedTempleVoteToken, ERC20Burnable) onlyAuthorized whenNotPaused {
+        _burn(account, value);
+    }
+
+    /**
+     * @dev See {IERC20-transferFrom}.
+     *
+     * Emits an {Approval} event indicating the updated allowance. This is not
+     * required by the EIP. See the note at the beginning of {ERC20}.
+     *
+     * NOTE: Does not update the allowance if the current allowance
+     * is the maximum `uint256`.
+     *
+     * Requirements:
+     *
+     * - `from` and `to` cannot be the zero address.
+     * - `from` must have a balance of at least `value`.
+     * - the caller must have allowance for ``from``'s tokens of at least
+     * `value`.
+     */
+    function transferFrom(address from, address to, uint256 value) public virtual override(ERC20, IERC20) whenNotPaused returns (bool) {
```

Comment:
Is `whenNotPaused` very useful here in that it's only authorized callers which are allowed to mint/burn/transfer anyway?

It appears that only the staking contract will be given this access, and that can also be paused. So this seems a bit redundant and adds a lot more LOC (granted mostly comments)